docs/src/main/paradox/couchbase.md

```diff
@@ -0,0 +1,173 @@
+#Couchbase
+
+The Couchbase connector allows you to read and write to Couchbase. You can query a bucket from CouchbaseSource using N1QL queries or use bulk operation to insert or update with CouchbaseFlow or CouchbaseSink. Couchbase connector uses Couchbase Java SDK behind the scene. 
+
+You can read more about Couchbase [here](https://www.couchbase.com/products/server) and you can read more about Couchbase Java SDK [here](https://developer.couchbase.com/documentation/server/current/sdk/java/start-using-sdk.html)
+
+Couchbase connector supports all Document formats which are supported by SDK. All those formats extends `Document<T>` interface and this is the level of abstraction that connector is using.
+
+## Artifacts
+
+@@dependency [sbt,Maven,Gradle] {
+  group=com.lightbend.akka
+  artifact=akka-stream-alpakka-couchbase_$scalaBinaryVersion$
+  version=$version$
+}
+
+#Usage
+
+We will need an @scaladoc[ActorSystem](akka.actor.ActorSystem) and an @scaladoc[ActorMaterializer](akka.stream.ActorMaterializer).
+
+Scala
+: @@snip (/couchbase/src/test/scala/docs/scaladsl/CouchbaseSupport.scala) { #init-actor-system }
+
+Java
+: @@snip (/couchbase/src/test/java/docs/javadsl/Examples.java) { #init-actor-system }
+
+
+
+Then we will need to connect to Couchbase cluster
+
+Scala
+: @@snip (/couchbase/src/test/scala/docs/scaladsl/CouchbaseSupport.scala) { #init-cluster }
+
+Java
+: @@snip (/couchbase/src/test/java/docs/javadsl/Examples.java) { #cluster-connect }
+
+#Source Usage
+
+We can create different types of `CouchbaseSource`.
+
+From Single Id
+
+Scala
+: @@snip (/couchbase/src/test/scala/docs/scaladsl/CouchbaseSupport.scala) { #init-sourceSingle }
+
+Java
+: @@snip (/couchbase/src/test/java/docs/javadsl/Examples.java) { #init-sourceSingle }
+ 
+From Bulk of Ids
+
+Scala
+: @@snip (/couchbase/src/test/scala/docs/scaladsl/CouchbaseSupport.scala) { #init-sourceBulk }
+
+Java
+: @@snip (/couchbase/src/test/java/docs/javadsl/Examples.java) { #init-sourceBulk }
+        
+
+From N1QL query
+
+Scala
+: @@snip (/couchbase/src/test/scala/docs/scaladsl/CouchbaseSupport.scala) { #init-sourcen1ql }
+
+Java
+: @@snip (/couchbase/src/test/java/docs/javadsl/Examples.java) { #init-sourcen1ql }
+
+#Write Operation Settings
+
+For each mutation operation we need to create `CouchbaseWriteSettings` instance which consists of the following parameters
+
+- Stage Parallelism (default 1)
+- Couchbase Replication Factor (default `ReplicateTo.None`) 
+- Couchbase Persistance Level for Write Operation (default `PersistTo.NONE`)
```

Comment:
Fo the Replication - let's change default value to One it means that we observe one replica of data besides the master and we will survive master replica failure. 

For the persistence - i would like to leave it as is because couchbase bucket divided to memory and disk. only  when bucket memory expands to some value (memory high watermark) it will start persistence to the disk in LRU manner. I would like not to enforce persistence and let couchbase decide when to persist this data.